### PR TITLE
Single select native

### PR DESCRIPTION
**Pull request by dawinValenzuela**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [x] New feature (non-breaking change which adds functionality)

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)

**Storybook Link** (Link to a local storybook url to review new functionality)
https://storybook-single-select-native-7v6rokdl5q-uc.a.run.app/?path=/story/composer-primitive-form-dropdown-single-select--basic

### CODE REVIEW

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        >
+          <option value="">{placeHolder}</option>
+          {options.map((item, index) => (
+            <option key={`${index}${item.value}`} value={item.value}>
+              {item.label}
+            </option>
+          ))}
+        </select>
+        <div className={iconWrapper}>
+          <Icon width={24} height={24} className={icon}>
```
Comment (guilean) [inline packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:94]: Remove width and height since these are default values for Icon

File: packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx [hunk 2]
```diff
@@ -61,34 +97,31 @@ export const DropdownSingleSelect = forwardRef<HTMLElement, Props>(
...
               >
-                {(selectedItem && <span>{selectedItem.label}</span>) || (
-                  <span className="text-outline-forms-filled">
-                    {placeHolder}
-                  </span>
-                )}
-                <Icon width={24} height={24} className="fill-current">
-                  <ExpandMore type="filled" />
+                <span>{selectedItem ? selectedItem.label : placeHolder}</span>
+                <Icon width={24} height={24} className={icon}>
```
Comment (guilean) [inline packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx:112]: Ditto

File: packages/components/src/styles/colors.js [hunk 3]
```diff
@@ -52,6 +52,8 @@ module.exports = {
     enabled: '#FFFFFF',
     focus: '#FFFFFF',
     disabled: '#D8D8D8',
+    inactive: 'rgba(0, 0, 0, 0.3)',
```
Comment (guilean) [inline packages/components/src/styles/colors.js:55]: Design said that `inactive` is the same as `enabled`. They have a missnaming on DSM. So we can remove this new inactive variable. 

We don't have to use fill-form colors for text color. This is a different property. Colors defined under fill-forms should be used only on the input box, not the content. We'll be using this fill-form property mainly on background colors.

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.test.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    render(
+      <DropdownNativeSelect
+        options={options}
+        onItemChange={onChange}
+        placeHolder="Placeholder"
+      />
+    );
+    screen.getByText('Placeholder');
+  });
+
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.test.tsx:28]: nit: please remove this empty lines between test cases, we are not adding them anywhere else

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import cn from 'classnames';
+import { Icon } from 'components/primitives';
+import { expand_more as ExpandMore } from 'components/primitives/Icon/icons/navigation';
+import { getDropdownSingleSelectClasses } from 'utils/getDropdownSingleSelectClasses';
+
+export const DROPDOWN_NATIVE_DEFAULT_TEST_ID = 'dropdown_native';
+
+export interface Option {
+  label: any;
+  value: any;
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:11]: can we receive any value for those properties?, we should avoid using any whenever possible

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+      initialSelectedOption,
+      onItemChange,
+      testId = DROPDOWN_NATIVE_DEFAULT_TEST_ID,
+      disabled = false,
+      name = 'dropdown',
+      invalid,
+    }: Props,
+    ref
+  ) => {
+    const defaultValue = initialSelectedOption && initialSelectedOption.value;
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:41]:
```suggestion
    const defaultValue = initialSelectedOption?.value;
```

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    const [itemSelected, setItemSelected] = useState(defaultValue || '');
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:42]: this is assuming that `defaultValue` is a string but the definition of the Option is of type `any`

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  placeHolder?: string;
+  initialSelectedOption?: Option;
+  onItemChange?: (item: Option) => void;
+  testId?: string;
+  disabled?: boolean;
+  name?: string;
+  invalid?: boolean;
+};
+
+export const DropdownNativeSelect = forwardRef<HTMLSelectElement, Props>(
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:26]: missing storybook for this component

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+      const index = event.target.selectedIndex;
+      const label = event.target[index].label;
+      event.target.className += ' !text-fill-forms-filled';
+
+      setItemSelected(event.target.value);
+
+      return onItemChange({
+        label: label,
+        value: event.target.value,
+      });
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:65]: can we just use?

```suggestion
      onItemChange(options[index]);
```

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+          )}
+          onChange={handleOnChange}
+          name={name}
+          disabled={disabled}
+          data-testid={testId}
+          ref={ref}
+        >
+          <option value="">{placeHolder}</option>
+          {options.map((item, index) => (
+            <option key={`${index}${item.value}`} value={item.value}>
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:88]: do we need to concatenate index and value, why not just use the index or the value?, actually this can produce duplicated keys if called with _malicious data_, for example if the item at index `1` has `01` as value and item at index `10` has `1` both will produce `101`.

File: packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.stories.tsx [hunk 11]
```diff
@@ -35,7 +35,7 @@ basic.story = {
 export const withInitialSelectedItem = () => {
   return (
     <DropdownSingleSelect
-      initialSelectedOption={options[3]}
+      initialSelectedOption={options[0]}
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.stories.tsx:38]: why this change?

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+      selectDisabledClasses,
+      selectErrorClasses,
+      iconWrapper,
+      icon,
+    } = getDropdownSingleSelectClasses(disabled, invalid, itemSelected);
+
+    const handleOnChange = (event) => {
+      const index = event.target.selectedIndex;
+      const label = event.target[index].label;
+      event.target.className += ' !text-fill-forms-filled';
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:58]: why do we need this, isn't the component updating itself with the appropriate styles?

File: packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx [hunk 13]
```diff
@@ -39,12 +40,47 @@ export const DropdownSingleSelect = forwardRef<HTMLElement, Props>(
     }: Props,
     ref
   ) => {
+    const defaultValue = initialSelectedOption && initialSelectedOption.value;
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx:43]:
```suggestion
    const defaultValue = initialSelectedOption?.value;
```

File: packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx [hunk 14]
```diff
@@ -39,12 +40,47 @@ export const DropdownSingleSelect = forwardRef<HTMLElement, Props>(
...
+      selectTextClasses,
+      selectFocusClasses,
+      selectDisabledClasses,
+      selectErrorClasses,
+      selectListClasses,
+      itemListClasses,
+      icon,
+    } = getDropdownSingleSelectClasses(disabled, invalid, itemSelected);
+
+    if (isMobile) {
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx:57]: please extract all the logic in this component to another file and in here just return a component or the other

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx (hunk 5 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:11]: @tkanzakic I think we have to let any, right now we don’t know what type could be this, because I saw on abstract some label with icons. https://share.goabstract.com/1e3aa0aa-fba8-4b47-aede-7ceb8fb6baeb?mode=design&sha=c8a4e5c55a37e48682b5f36f87dc448ae1a876ca

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.test.tsx (hunk 4 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.test.tsx:28]: Done!

File: packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx (hunk 2 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx:112]: Done!

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx (hunk 1 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:94]: Done!

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx (hunk 10 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:88]: Done!

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx (hunk 12 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:58]: You're right, I remove line 58.

File: packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.stories.tsx (hunk 11 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.stories.tsx:38]: revert that change.

File: packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx (hunk 13 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx:43]: done

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx (hunk 6 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:41]: Done!

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx (hunk 7 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:42]: For now, I change the label to a string, we need to validate with the design team wich values could take this dropdown, for example, I saw a list with icon and label.

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx (hunk 8 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:26]: Done!

File: packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx (hunk 9 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownNativeSelect/DropdownNativeSelect.tsx:65]: Done!

File: packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx (hunk 14 above)
Comment (dawinValenzuela) [reply packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx:57]: Done!

File: packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.test.tsx [hunk 15]
```diff
@@ -1,146 +1,15 @@
...
-        onItemChange={onChange}
-        placeHolder="Placeholder"
-        invalid
-      />
-    );
-    const dropdown = screen.getByTestId(
-      DROPDOWN_DEFAULT_TEST_ID
-    ) as HTMLButtonElement;
-    expect(dropdown).toHaveClass('border-fill-system-error');
+  it('should render downshift select for desktop version', () => {
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.test.tsx:11]: missing test case for the mobile branch

File: packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx [hunk 16]
```diff
@@ -24,83 +22,10 @@ type Props = {
...
-                </ul>
-              )}
-            </div>
-          );
-        }}
-      </Downshift>
-    );
-  }
-);
+export const DropdownSingleSelect: FunctionComponent<SelectProps> = (props) => {
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx:25]: should this be wrapped by a `forwardRef`?
